### ISSUE TITLE
BRD - Emperor Dagran Thaurissan Incorrect/Missing Quotes

### ISSUE THREAD

**Issue by Azcobu** [azerothcore/azerothcore-wotlk#7726]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [X] 50-59

### Current Behaviour
Original report: https://github.com/chromiecraft/chromiecraft/issues/1630

Emperor Dagran Thaurissan does not use his quotes when clearing the Senate. The only thing he says is "Come to aid the Throne!" when aggroed, and he says that whether the Senate is alive or dead.

Tester: confirmed
- While clearing the room, he doesn't react at all, when he should be saying the quotes under 'Fighting the Senate' below. 
- Getting Emperor Dagran Thaurissan into combat when there are mobs alive in the Senate does make Dragan say his quote ("Come to aid the Throne!") properly. 
- Getting Emperor Dagran Thaurissan into combat when the Senate is dead fails because he says "Come to aid the Throne!" anyway, and not the correct "Thank you for clearing out those foolish senators. Now prepare to meet your doom at the hands of Ragnaros' most powerful servant." 

Room cleared, not a word out of him:
![WoWScrnShot_090521_115748](https://user-images.githubusercontent.com/81782124/132113760-4ef4c15a-3941-424f-9c14-0994dc8dc88f.jpg)

- Getting a player killed doesn't make him say " Hail to the king, baby!":

![WoWScrnShot_090521_122628](https://user-images.githubusercontent.com/81782124/132113770-b04cde57-309d-4b11-8026-50e3a7cfa618.jpg)

Also noticed you can pull Dagran and Moira just stands there instead of assisting:

![WoWScrnShot_090521_120218](https://user-images.githubusercontent.com/81782124/132113767-cfff304f-7acf-4766-9918-912f8710eae4.jpg)

When Dagran dies, Moira should also play an emote: "is visibly shaken by the death of Emperor Thaurissan."

### Expected Blizzlike Behaviour

![image](https://user-images.githubusercontent.com/22718174/132107429-c6dd517c-4a56-492d-a08a-035bc8bcc4e6.png)

### Source
I was actually a bit skeptical about this because the sources provided by the reporter are much more modern than Wrath, and the contemporary sources only list one or two quotes. However, TBC Wowhead does show all the quotes below on this NPC's page here: https://tbc.wowhead.com/npc=9019/emperor-dagran-thaurissan

https://wowpedia.fandom.com/wiki/Dagran_Thaurissan

### Steps to reproduce the problem

.go c 90614 to get into the room, then murder NPCs.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/595bb6adccbabc714469f3935541978283b8bdfb

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-corldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by Asthalor** [comment]
The quotes can be added to the creature with the following code, but i dont know C++ to implement those ids to the code.
Some of them need to be told when creatures die next to him (next i say by 200 yards radious) and the other is a change on the aggro line when no creatures are left on the throne room.

```
-- Inserting lines for thaurrisan
DELETE FROM `creature_text` WHERE `CreatureID` = 9019;
INSERT INTO `creature_text` (`CreatureID`,`GroupID`,`ID`,`Text`,`Type`,`Language`,`Probability`,`Emote`,`Duration`,`Sound`,`BroadcastTextId`,`TextRange`,`COMMENT`) VALUES
(9019,0,0,'Come to aid the Throne!',14,0,100,0,0,0,0,0,'Aggro line council alive'),
(9019,1,0,'I will crush you into little tiny pieces!',14,0,100,0,0,0,5457,0,'Aggro line council dead'),
(9019,2,0,'Hail to the king, baby!',14,0,100,0,0,0,5431,0,'Killing player'),
(9019,3,0,'Your efforts are utterly pointless, fools!  You will never be able to defeat me!',14,0,100,0,0,0,5312,0,'Killing council 1'),
(9019,4,0,'They were just getting in the way anyways.',14,0,100,0,0,0,5313,0,'Killing council 2'),
(9019,5,0,'Ha! You can''t even begin to imagine the futility of your efforts.',14,0,100,0,0,0,5314,0,'Killing council 3'),
(9019,6,0,'Is that the best you can do?  Do you really expect that you could defeat someone as awe inspiring as me?',14,0,100,0,0,0,5315,0,'Killing council 4'),
(9019,7,0,'Thank you for clearing out those foolish senators.  Now prepare to meet your doom at the hands of Ragnaros'' most powerful servant.',14,0,100,0,0,0,5315,0,'Killing all council');
```

**Comment by patou01** [comment]
Could you clarify when the yells are supposed to happen? Is it based on certain mobs, randomly, certain packs? I'm reworking a big chunk of the dungeon at the moment and this sounds like something I want to include.

**Comment by patou01** [comment]
I have made this somewhat working with the following principle, but I'm not happy with it.
When the creatures are created, check their distance to the boss.
Assign one of the senators in that range to a yell.
For instance, the first mobs are between 90 and 120 yards from the boss. The first senator that is spawned will link to the 1st yell.

However, this means that depending on spawn order and other potential things, you might clear an entire side and get no yells, and then get all the yells at once on the other side. 

So I have 2 questions:
**Should I make an implementation based on the proportion of senators that have died instead?**
**is the last yell done when the room is fully cleared or when the last senator dies?**

**Comment by Azcobu** [comment]
Looking at videos of this, my general impression was that the council death yells were fairly randomly distributed. So a simpler implementation might be give each senator's death an X% chance of triggering one of the four 'council death' yells. You might need to experiment a bit to find a good value for X - maybe about 10%? That's just a suggestion thogh, I don't know if it would be easier to implement.

For your second question, I think it makes more sense to trigger on the last senator's death, rather than from an Twilight Ambassador or some other NPC that's not a senator.

**Comment by patou01** [comment]
Both are possible, I feelthat adding RNG could mean you get a room with no sound and suddenly he starts screaming in your ears all the time since he's expected to do 5 yells? Or is the logic "5 is a maximum, but it could be less" ?

I'm trying to make an implementation based on the proportion of senators that have died at the moment. So every 20% of them that dies triggers a new yell, with the last yell happening on 100%.

**Comment by Azcobu** [comment]
I think the logic was there's 4 yells he could choose from, and he would randomly yell one every now and then as mobs died. He repeats himself in some videos I've seen. (See https://youtu.be/2mIKQwgoBT8?t=1467 for one example showing repeated lines.)
Your approach is arguably better, in that there's no repetitions and you'd see all 5 lines eventually.

**Comment by patou01** [comment]
Thanks. I don't like when mobs yell the same thing repeatedly, it sounds weird to me when it happens. Though with the spread we have now in the room you get more yells at start I think, but I think that's a separate issue where the spawns are maybe not 100% correct.

The proportional one seems kinda okay when blasting through the mobs. I'll make a PR for it next week probably, but it will come with a lot of other changes (unfortunately I got tangled into the "let me also fix this thing" mindset so by now I've reworked 3 or 4 events in BRD on the same branch...)
